src/AppInstallerRepositoryCore/Manifest/ManifestValidation.cpp

```diff
@@ -6,126 +6,125 @@
-                if (!processedFields.insert(fieldInfo.Name).second)
-                {
-                    errors.emplace_back(ManifestError::FieldDuplicate, fieldInfo.Name, "", keyValuePair.first.Mark().line, keyValuePair.first.Mark().column);
-                }
+            if (in1.Scope != in2.Scope)
+            {
+                return in1.Scope < in2.Scope;
+            }
 
-                // Validate non empty value is provided for required fields
-                if (fieldInfo.Required)
-                {
-                    if (!valueNode.IsDefined() || valueNode.IsNull() ||  // Should be defined and not null
-                        (valueNode.IsScalar() && valueNode.as<std::string>().empty()) ||  // Scalar type should have content
-                        ((valueNode.IsMap() || valueNode.IsSequence()) && valueNode.size() == 0))  // Map or sequence type should have size greater than 0
-                    {
-                        errors.emplace_back(ManifestError::RequiredFieldEmpty, fieldInfo.Name, "", valueNode.Mark().line, valueNode.Mark().column);
-                    }
-                }
+            return false;
+        };
 
-                // Validate value against regex if applicable
-                if (fullValidation && !fieldInfo.RegEx.empty())
-                {
-                    std::string value = valueNode.as<std::string>();
-                    std::regex pattern{ fieldInfo.RegEx };
-                    if (!std::regex_match(value, pattern))
-                    {
-                        errors.emplace_back(ManifestError::InvalidFieldValue, fieldInfo.Name, value, valueNode.Mark().line, valueNode.Mark().column);
-                        continue;
-                    }
-                }
+        std::set<ManifestInstaller, decltype(installerCmp)> installerSet(installerCmp);
+        bool duplicateInstallerFound = false;
+
+        // Validate installers
+        for (auto const& installer : manifest.Installers)
+        {
+            if (!duplicateInstallerFound && !installerSet.insert(installer).second)
+            {
+                resultErrors.emplace_back(ManifestError::DuplicateInstallerEntry);
+                duplicateInstallerFound = true;
+            }
 
-                if (!valueNode.IsNull())
+            if (installer.Arch == Utility::Architecture::Unknown)
+            {
+                resultErrors.emplace_back(ManifestError::InvalidFieldValue, "Arch");
+            }
+
+            if (installer.InstallerType == ManifestInstaller::InstallerTypeEnum::Unknown)
+            {
+                resultErrors.emplace_back(ManifestError::InvalidFieldValue, "InstallerType");
+            }
+
+            if (installer.InstallerType == ManifestInstaller::InstallerTypeEnum::MSStore)
+            {
+                // MSStore type is not supported in community repo
+                resultErrors.emplace_back(
+                    ManifestError::FieldNotSupported, "InstallerType",

```

Comment:
>ManifestError::FieldNotSupported, "InstallerType", [](start = 20, length = 50)

Probably need ValueNotSupported so that it can be more clear that it isn't InstallerType, but it's value that is bad.